### DISCUSSION TITLE
problem building the out of the tree package msprime

### DISCUSSION THREAD

**Discussion by JoseBlanca** [pyodide/pyodide#4173] [General]
Hi,

I'm trying to build msprime (https://github.com/tskit-dev/msprime).
I get the following error:

$ pyodide build
(...)
/tmp/tmp10v480nq/cc -DNDEBUG -g -fwrapv -O3 -Wall -O2 -g0 -fPIC -DPY_CALL_TRAMPOLINE -Ilwt_interface -Ilib -Igit-submodules/tskit/c -Igit-submodules/tskit/c/tskit -Igit-submodules/tskit/c/subprojects/kastore -I/usr/include -I/tmp/build-env-k71qr6n5/lib/python3.11/site-packages/numpy/core/include -I/tmp/build-env-k71qr6n5/include -I/home/jose/.pyenv/versions/3.11.5/include/python3.11 -c lib/avl.c -o build/temp.emscripten_3_1_45_wasm32-cpython-311/lib/avl.o -std=c99
/tmp/tmp10v480nq/cc -DNDEBUG -g -fwrapv -O3 -Wall -O2 -g0 -fPIC -DPY_CALL_TRAMPOLINE -Ilwt_interface -Ilib -Igit-submodules/tskit/c -Igit-submodules/tskit/c/tskit -Igit-submodules/tskit/c/subprojects/kastore -I/usr/include -I/tmp/build-env-k71qr6n5/lib/python3.11/site-packages/numpy/core/include -I/tmp/build-env-k71qr6n5/include -I/home/jose/.pyenv/versions/3.11.5/include/python3.11 -c lib/fenwick.c -o build/temp.emscripten_3_1_45_wasm32-cpython-311/lib/fenwick.o -std=c99
lib/fenwick.c:31:10: fatal error: 'gsl/gsl_math.h' file not found
   31 | #include <gsl/gsl_math.h>

I thinks that this issue was discussed already and a fix was proposed in here: https://github.com/pyodide/pyodide/pull/2548

I have tried to fix the issue by setting the FLAGS in this way:

$ export CFLAGS=$(pkg-config --cflags gsl)
$ export LDFLAGS=$(pkg-config --libs gsl)
$ echo $LDFLAGS
-lgsl -lgslcblas -lm

After that I have tried to build the package using pip build:

$ pip install build

This has worked just fine, but still pyodide build fails. I don't know how to fix the issue. Any help would be much appreciated.
Best,

Jose Blanca

**Comment by ryanking13** [comment]
Unlike in-tree build, out-of-tree build does not handle host dependencies. So you need to cross-compile `libgsl` to WASM manually and pass the correct include / library path (CFLAGS, LDFLAGS) to the built libgsl when building msprime.
